transports/quic/src/endpoint.rs

```diff
@@ -379,6 +370,13 @@ async fn background_task(
                         let is_drained_event = event.is_drained();
                         if is_drained_event {
                             alive_connections.remove(&connection_id);
+                            if is_orphaned && alive_connections.is_empty() {
+                                tracing::info!(
+                                    "Listener closed and no active connections remain. Shutting down the background task."
+                                );
```

Comment:
Does this affect the invariant you expressed with an `unreachable` in d46b72e9af0a85038d5f79cc6b877eed61fe8109?

Reply:
Nope. The path marked as `unreachable` concerns the channel that is used from the `Endpoint` struct to the background task and would only be reachable if the channel is explicitly closed or all senders dropped.  However, nowhere in the code we explicitly close that channel, and because the background task itself holds one `Endpoint` (so it can clone it and provide it to new connections) it can never happen that all senders are dropped. Does that make sense? Can this as a comment.

Reply:
Yep that makes sense, thank you!